website/database.py

```diff
@@ -775,12 +779,21 @@ def get_quiz_stats(self, ids, start=None, end=None):
 
     def add_program_stats(self, id, level, number_of_lines, exception):
         key = {"id#level": f"{id}#{level}", "week": self.to_year_week(date.today())}
-
         add_attributes = {"id": id, "level": level, "number_of_lines": number_of_lines}
+        p_stats = PROGRAM_STATS.get_many({"id": id, "week": self.to_year_week(date.today())})
+
+        chart_history = []
+        if p_stats.records and 'chart_history' in p_stats.records[0].keys():
+            chart_history = p_stats.records[0]['chart_history']
```

Comment:
This is not very defensive (if there is no 'chart_history' it will fail hard) can we change it into a get?

Reply:
I didn't write the code, but doesn't the if statement above check for the key existing though? Or perhaps I don't understand what you mean exactly. But I will change it into a get, that's nicer anyway :)